### PR TITLE
Bump mysql-connector-java to 8.0 to support current MySQL version

### LINKED ISSUE #766: outdated MySQL driver in elide-standalone and examples
MySQL changed default auth method in 8.0, elide examples crash when trying to connect to modern MySQL instances
https://dev.mysql.com/doc/refman/8.0/en/mysql-nutshell.html
> The caching_sha2_password and sha256_password authentication plugins provide more secure password encryption than the mysql_native_password plugin, and caching_sha2_password provides better performance than sha256_password. Due to these superior security and performance characteristics of caching_sha2_password, it is now the preferred authentication plugin, and is also the default authentication plugin rather than mysql_native_password. For information about the implications of this change of default plugin for server operation and compatibility of the server with clients and connectors, see caching_sha2_password as the Preferred Authentication Plugin.

### PR DESCRIPTION
**Pull request by harshtuna**
Fixes #766

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by harshtuna** [comment]
> Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! 😄

done

**Comment by aklish** [comment]
Changes look good.  Can you update the changelog.md?  Thx.

**Comment by harshtuna** [comment]
hold on. merged with master but there's PR #770 on the way
merge that one first, I'll sync with master again

File: pom.xml [hunk 1]
```diff
@@ -145,6 +146,11 @@
                 <artifactId>jetty-servlet</artifactId>
                 <version>${version.jetty}</version>
             </dependency>
+            <dependency>
+                <groupId>mysql</groupId>
+                <artifactId>mysql-connector-java</artifactId>
+                <version>${version.mysql}</version>
```
Comment (wcekan) [inline pom.xml]: Should this not still default to `<scope>test</scope>` unless overridden?  Only the hibernate connectors require this dependency at runtime.

Comment (harshtuna) [reply pom.xml]: No component is using it for testing as of today. Dependency scope:
* `elide-blog-example` - runtime
* `elide-blog-example-resteasy` - runtime
* `elide-hibernate3-mysql-example` - runtime
* `elide-standalone` - compile

And in `elide-standalone` tests are actually using `InMemoryDataStore`

**Comment by harshtuna** [comment]
merge after pr #771

File: pom.xml (hunk 1 above)
Comment (harshtuna) [reply pom.xml]: Note - `jetty-server` and `jetty-servlet` have similar usage and are not in test scope

**Pull request merged**